src/js/libcs/Accessors.js

```diff
@@ -168,6 +168,22 @@ Accessor.setField = function(geo, field, value) {
         movepointscr(geo, List.turnIntoCSList([value.value[0], value.value[1], CSNumber.real(1)]), "homog");
     }
 
+    if (field === "x" && geo.kind === "P" && geo.movable && value.ctype === "number") {
+        if (CSNumber._helper.isAlmostZero(geo.homog.value[2])) {
+            console.log("cannot set x component without y component of far points");
+        } else {
+            movepointscr(geo, List.turnIntoCSList([value, CSNumber.div(geo.homog.value[1], geo.homog.value[2]), CSNumber.real(1)]), "homog");
+        }
+    }
```

Comment:
@kortenkamp In the original code, the case distinction precludes assignment of `[x, NaN, 1]`.

If we drop the case distinction, we allow the user to assign `NaHC` as a homogeneous coordinate of a point. Otherwise, such an assignment is not possible (at least using this method).

I think simply ignoring a requested assignment may have at least as unexpected consequences as assigning `NaHC`. In this case, the warning may be justified. (We also still warn about division by zero in `CSNumber.inv`.)

An assignment of `NaHC` can be detected by users without a log message. This seems much cleaner. (If users *want* to shoot themselves in the foot by assigning xy-coordinates of far points, why should we not let them?)

I am in favour of dropping the case distinction as @gagern suggested.